packages/js/data/src/products/selectors.ts

```diff
@@ -145,6 +147,29 @@ export const getPermalinkParts = createSelector(
 	}
 );
 
+/**
+ * Returns an array of related products for a given product ID.
+ *
+ * @param {ProductState} state     - The current state.
+ * @param {number}       productId - The product ID.
+ * @return {PartialProduct[]}        The related products.
+ */
+export const getRelatedProducts = (
+	state: ProductState,
+	productId: number
+) => {
+	const product = state.data[ productId ];
+	if ( ! product?.related_ids ) {
+		return [];
```

Comment:
For this and line 170, I would either create a constant outside of this function `const EMPTY_ARRAY = []` and use that, or make use of the `createSelector` method used for the `getProducts` selector as well.

Reason for doing so, is that if you use this selector within a `useSelect` and no related ids exist it will generate a new array object every time the function is called, which may break any memoization in the `useSelect`.
Maybe the return of the `const` would be the best approach?

Reply:
https://github.com/woocommerce/woocommerce/pull/43489/commits/854d02afea54c6c7fd1bfcbea9a5f918bc2e81ab